src/mgr/DaemonServer.cc

```diff
@@ -1643,20 +1643,23 @@ bool DaemonServer::_handle_command(
 	    continue;
 	  }
 	  touched_pgs++;
-	  if (!(q.second.state & PG_STATE_ACTIVE) ||
-	      (q.second.state & PG_STATE_DEGRADED)) {
-	    ++dangerous_pgs;
-	    continue;
-	  }
+
 	  const pg_pool_t *pi = osdmap.get_pg_pool(q.first.pool());
 	  if (!pi) {
 	    ++dangerous_pgs; // pool is creating or deleting
-	  } else {
-	    if (pg_acting.size() < pi->min_size) {
-	      ++dangerous_pgs;
-	    }
+            continue;
 	  }
-	}
+
+	  if (!(q.second.state & PG_STATE_ACTIVE) ||
+              ((pi->type != pg_pool_t::TYPE_REPLICATED) &&
+               (q.second.state & PG_STATE_DEGRADED))) {
+	    ++dangerous_pgs;
+	    continue;
+	  }
+          if (pg_acting.size() <= pi->min_size) {
```

Comment:
this changed from ``<`` to ``<=``.. was that intentional?  I think ``<`` is correct...

Reply:
If pg_acting.size() == pi->min_size then wouldn't write access be lost of another OSD is stopped?  Isn't that the point of ok-to-stop checking?

Reply:
pg_acting is the anticipated acting set if the osd(s) are stopped (current acting, minus any of the osds provided as an argument)

Reply:
@xxhdx1985126 Change this back to pg_acting.size() < pi->min_size